src/coreclr/classlibnative/bcltype/arraynative.cpp

```diff
@@ -1153,3 +1153,40 @@ FCIMPL2_IV(void, ArrayNative::InitializeArray, ArrayBase* pArrayRef, FCALLRuntim
     HELPER_METHOD_FRAME_END();
 }
 FCIMPLEND
+
+FCIMPL4(void, ArrayNative::GetSpanDataFrom, FCALLRuntimeFieldHandle structField, FCALLRuntimeTypeHandle targetType, void** data, INT32* count)
+{
+    FCALL_CONTRACT;
+    struct
+    {
+        REFLECTFIELDREF refField;
+    } gc;
+    gc.refField = (REFLECTFIELDREF)ObjectToOBJECTREF(FCALL_RFH_TO_REFLECTFIELD(structField));
+    HELPER_METHOD_FRAME_BEGIN_PROTECT(gc);
+
+    FieldDesc* pField = (FieldDesc*)gc.refField->GetField();
+
+    if (!pField->IsRVA())
+        COMPlusThrow(kArgumentException);
+
+    TypeHandle targetTypeHandle = FCALL_RTH_TO_REFLECTCLASS(targetType)->GetType();
+    if (!CorTypeInfo::IsPrimitiveType(targetTypeHandle.GetSignatureCorElementType()) && !targetTypeHandle.IsEnum())
+        COMPlusThrow(kArgumentException);
+
+    DWORD totalSize = pField->LoadSize();
+    DWORD targetTypeSize = targetTypeHandle.GetSize();
+
+    // Report the RVA field to the logger.
+    g_IBCLogger.LogRVADataAccess(pField);
+
+    _ASSERTE(data != NULL && count != NULL);
+    *data = pField->GetStaticAddressHandle(NULL);
+    *count = (INT32)totalSize / targetTypeSize;
+
+#if BIGENDIAN
+    COMPlusThrow(kPlatformNotSupportedException);
```

Comment:
Do we want to do anything more than this?

Reply:
Does this mean the API throws on BigEndian?

Reply:
Yes.

Reply:
This is CoreCLR specific path. CoreCLR does not support any big endian platforms today. This PlatformNotSupportedException is more like TODO in case CoreCLR gets enabled on a big endian platform.

The same path on Mono has proper handling of big endian platforms.

Reply:
>CoreCLR does not support any big endian platforms today.

@jkotas Isn't the s390x big endian?

Reply:
Yes, it is. s390x port is Mono only.

Reply:
> Yes, it is. s390x port is Mono only.

Oh. I didn't realize that. Thanks.